smartsim/_core/launcher/step/localStep.py

```diff
@@ -37,8 +37,13 @@ class LocalStep(Step):
     def __init__(self, name: str, cwd: str, run_settings: RunSettings):
         super().__init__(name, cwd, run_settings)
         self.run_settings = run_settings
-        self.env = self._set_env()
+        self._env = self._set_env()
 
+    @property
+    def env(self) -> t.Dict[str, str]:
```

Comment:
I notice that this override returns a `dict` but the base class has `optional[dict]`. This seems like it might break the LSP and result in potential null object errors.

Should we make them consistent?

Reply:
It should be okay for a method on a subclass to return a subclass of the expected type. All instances of type `T` are implicitly `T | None` as well (they just happen to never be `None`).

I chose to narrow the type for mainly as a convenience for developers who happen to know for fact they are working with a `LocalStep`. For e.g.
```py
step: Step = ...
step.env.items()  # <-- mypy should error here saying "type `None` has no method `items`"

known_local_step: LocalStep = ...
known_local_step.env.items()  # <-- mypy is happy bc it knows that this is fine
                              #      and devs do not need to make a "is not none" check here
 ```

This would break and going the other direction though (returning a `T | None` where a `T` is expected) and mypy _should_ be able to report that error.

Reply:
Great! I forgot that mypy would save me from myself. TYS!